packages/orchestration/src/orchestration-client.ts

```diff
@@ -168,30 +169,34 @@ export class OrchestrationClient {
   private parseAndMergeTemplating(
     config: OrchestrationModuleConfig
   ): OrchestrationModuleConfig {
+    let parsedObject;
+    if (typeof config.templating === 'string' && !config.templating.trim()) {
+      throw new Error('Templating must be a non-empty YAML string.');
+    }
     try {
-      const parsedObject = yaml.parse(config.templating as string); // We are sure it's a string here
-      const result = promptTemplatePostRequestSchema.safeParse(parsedObject);
-
-      if (result.success) {
-        return {
-          ...config,
-          templating: {
-            template: result.data.spec.template,
-            ...(result.data.spec.defaults && {
-              defaults: result.data.spec.defaults
-            }),
-            ...(result.data.spec.response_format && {
-              response_format: result.data.spec.response_format
-            }),
-            ...(result.data.spec.tools && { tools: result.data.spec.tools })
-          }
-        };
-      }
-      throw new Error(
-        `Prompt Template YAML does not conform to the defined type: ${result.error}`
-      );
+      parsedObject = yaml.parse(config.templating as string);
     } catch (error) {
       throw new Error(`Error parsing YAML: ${error}`);
     }
+
+    const result = promptTemplatePostRequestSchema.safeParse(parsedObject);
+    if (!result.success) {
+      throw new Error(
+        `Prompt Template YAML does not conform to the defined type. Validation errors: ${result.error}`
+      );
+    }
+    return {
+      ...config,
+      templating: {
+        template: result.data.spec.template,
+        ...(result.data.spec.defaults && {
+          defaults: result.data.spec.defaults
+        }),
+        ...(result.data.spec.response_format && {
+          response_format: result.data.spec.response_format
+        }),
+        ...(result.data.spec.tools && { tools: result.data.spec.tools })
```

Comment:
[pp] Looks like we can extract `result.data.spec` to make the code more readable.